cmd/check_performance_counter/cmd/root.go

```diff
@@ -27,27 +31,47 @@ The defaults for this check have the --critical and --warning values set to 0, a
 retrieved is compared to be lesser than those values. Generally a counter value will be > 0, causing
 this check to emit an OK response when using these defaults.`,
 		Run: func(cmd *cobra.Command, args []string) {
-			cmd.ParseFlags(os.Args)
+			// cmd.ParseFlags(os.Args)
 
-			msg, retval := nagiosfoundation.CheckPerformanceCounter(warning, critical, greaterThan, pollingAttempts,
-				pollingDelay, metricName, counterName)
-
-			fmt.Println(msg)
-			os.Exit(retval)
+			if len(counterName) != 0 && !(len(counterName) > 1 && (metricName != "" || warning != 0 || critical != 0 || greaterThan)) {
+				var empty_cntr bool
+				empty_cntr = false
+				for _, counter := range counterName {
+					counter = strings.TrimSpace(counter)
+					if len(counter) == 0 {
+						empty_cntr = true
+						continue
+					}
+					msg, retval := nagiosfoundation.CheckPerformanceCounter(warning, critical, greaterThan, pollingAttempty_cntrs,
+						pollingDelay, metricName, counter)
+					fmt.Println(msg)
+					if finalretval < retval {
+						finalretval = retval
+					}
+				}
+				if empty_cntr {
+					fmt.Println("Input given is an empty counter, pass a valid counter name")
+					os.Exit(2)
+				}
+				os.Exit(finalretval)
+			} else {
+				fmt.Println("When passing multiple counters to the performance counter -m(--metricname) -w(--warning) -c(--critical) -g(greaterthan) flags cannot be used.")
+				os.Exit(2)
+			}
 		},
 	}
 
 	initcmd.AddVersionCommand(rootCmd)
 
 	const counterNameFlag = "counter_name"
-	rootCmd.Flags().StringVarP(&counterName, counterNameFlag, "n", "", "the name of the performance counter to check")
+	rootCmd.Flags().StringSliceVarP(&counterName, counterNameFlag, "n", nil, "the name of the performance counter to check")
 	rootCmd.MarkFlagRequired(counterNameFlag)
 	rootCmd.Flags().StringVarP(&metricName, "metric_name", "m", "", "the name of the metric generated by this check")
-	rootCmd.Flags().Float64VarP(&warning, "warning", "w", 0, "the threshold to issue a warning alert")
-	rootCmd.Flags().Float64VarP(&critical, "critical", "c", 0, "the threshold to issue a critical alert")
+	rootCmd.Flags().Float64VarP(&warning, "warning", "w", 0.0, "the threshold to issue a warning alert")
+	rootCmd.Flags().Float64VarP(&critical, "critical", "c", 0.0, "the threshold to issue a critical alert")
 	rootCmd.Flags().BoolVarP(&greaterThan, "greater_than", "g", false, "issue warnings if the metric is greater than the expected thresholds (default false)")
-	rootCmd.Flags().IntVarP(&pollingAttempts, "polling_attempts", "a", 2, "the number of times to fetch and average the performance counter")
-	rootCmd.Flags().IntVarP(&pollingDelay, "polling_delay", "d", 1, "the number of seconds to delay between polling attempts")
+	rootCmd.Flags().IntVarP(&pollingAttempty_cntrs, "polling_attempty_cntrs", "a", 2, "the number of times to fetch and average the performance counter")
```

Comment:
why are we renaming the variable here?  id love to leave the interface unchanged so I can avoid a major version upgrade

Reply:
This is actually done by mistake john, any change is not required in this line. I will update the PR